### PR TITLE
BUGFIX: Make Array.push accept `null` for array type casting

### PR DESCRIPTION
**Pull request by mhsdesign**
related: https://github.com/neos/neos-development-collection/pull/3658
fixes: https://github.com/neos/neos-development-collection/issues/3657

### CODE REVIEW

File: Neos.Eel/Classes/Helper/ArrayHelper.php [hunk 1]
```diff
@@ -351,11 +351,11 @@ public function push($array, $element): array
         if ($array instanceof \Traversable) {
             $array = iterator_to_array($array);
         }
-        if (is_scalar($array)) {
+        if (is_scalar($array) || $array === null) {
             $array = [$array];
         }
```
Comment (kdambekalns) [inline Neos.Eel/Classes/Helper/ArrayHelper.php:356]: I did this locally:

```suggestion
        if (is_scalar($array)) {
            $array = [$array];
        }
        if ($array === null) {
            $array = [];
        }
```

That way NULL does not end up in the array. This is more similar to the code replaced with #3654, where NULL would be skipped.

Comment (jonnitto) [reply Neos.Eel/Classes/Helper/ArrayHelper.php:356]: I would also prefer @kdambekalns solution

**Review by kdambekalns — APPROVED** [review]
I like it! 😎

**Comment by mhsdesign** [comment]
lol i was on this the same second ^^

**Review by jonnitto — APPROVED** [review]

**Comment by kdambekalns** [comment]
Thanks for fixing the test, @jonnitto!

**Pull request merged**